src/snowflake/snowpark/modin/pandas/dataframe.py

```diff
@@ -1813,12 +1813,31 @@ def unstack(self, level=-1, fill_value=None):  # noqa: PR01, RT01, D200
                 query_compiler=self._query_compiler.unstack(level, fill_value)
             )
 
-    @dataframe_not_implemented()
-    def pivot(self, index=None, columns=None, values=None):  # noqa: PR01, RT01, D200
+    def pivot(
+        self,
+        *,
+        columns: Any,
+        index: Any | NoDefault = no_default,
+        values: Any | NoDefault = no_default,
```

Comment:
Does values have to be hashable? Or are we pretty sure it is hashable already?

Reply:
On the pandas docs, they describe values as `str, object or a list of the previous`. So in theory you could have an object passed that is not hashable.

In `pivot_table`, we currently check whether Index, Columns, and Values are hashable and if not raise a NotImplementedError. Currently, we don't support anything other than strings or list of strings at the moment but have this TODO marked in the code: `TODO: SNOW-857485 Support for non-str and list of non-str for index/columns/values`